### ISSUE TITLE
Quick Question - How to to switch positions of two rows

### ISSUE THREAD

**Issue by andkov** [IHACRU/ihacru-3t-mhsu#1]
```r
library(magrittr)

```
Consider a data set `ds`, defined as:
```r
ds <- data.frame(
  "a" = c(2,0,5,0),
  "b" = c(5,0,3,3),
  "c" = c(9,0,5,1)
)
ds

> ds
  a b c
1 2 5 9
2 0 0 0
3 5 3 5
4 0 3 1
```
We would like to compose a general function to switch positions of any two given rows:
```r
> switch_two_rows <- function(
+   d     # data sets in which to perform the operation 
+   ,row1 # one of the rows to switch
+   ,row2 # another of the rows to switch
+ ){
+   # extract
+   d_row1 <- d[row1,]
+   d_row2 <- d[row2,]
+   # overwrite
+   d[row1,] <- d_row2
+   d[row2,] <- d_row1
+   
+   return(d)
+ }
```
Now we demonstrate the performance of the function:
```r
> ds
  a b c
1 2 5 9
2 0 0 0
3 5 3 5
4 0 3 1

> ds %>% switch_two_rows(2,4)
  a b c
1 2 5 9
2 0 3 1
3 5 3 5
4 0 0 0
```

**Comment by andkov** [comment]
@aetiologicCanada , you are correct: I would prefer to rely on a explicitly stated order of factor levels as the means to control the order in which values are presented during display.  However, Ernie asked me how to perform that operation in R. I couldn't understand the purpose either, but then I thought, hey that would be an interesting exercise in script writing. Notice, for example that it does not contain a single superfluous character and the names of the objects were chosen to have a linguistic continuity, like
```r
d_row1 <- d[row1,]
```
Also notice explicit documentation for the function input, as well as indented comments. 
```r
   d     # data sets in which to perform the operation 
   ,row1 # one of the rows to switch
   ,row2 # another of the rows to switch
```

My intent was to practice writing  R scripts that would be a good communication/learning tool for a general audience.  So in this case, we have an example of a trivial operation (which we could also achieve by certain matrix operations, but then we need to convert them into matrices, which has its own problems, so I chose not to take that path). So the value is not in achieving the end result of the operation, but rather on how to learn to design a function that would perform some operation. 

It took me about 15 minutes and was a great exercise. 

Thanks for asking!

**Comment by aetiologicCanada** [comment]
Cool. Forcats:: might have relevant functions for the general case
